### PR TITLE
feat(pgo): bash pgo for the win

### PR DESCRIPTION
**Pull request by lseman**

### CODE REVIEW

File: bash-pgo/PKGBUILD [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+# Contributor:  Bartłomiej Piotrowski <bpiotrowski@archlinux.org>
+# Contributor: Allan McRae <allan@archlinux.org>
+# Contributor: Aaron Griffin <aaron@archlinux.org>
+# PGO Version: Laio Seman <laio [at] ieee.org>
+
+pkgname=bash
+_basever=5.2
+_patchlevel=021
+pkgver=${_basever}.${_patchlevel}
+pkgrel=1
```
Comment (ptr1337) [inline bash-pgo/PKGBUILD:12]: Bump pkgrel to 2

File: bash-pgo/PKGBUILD [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+build() {
+    cd $pkgname-$_basever
+
+    _bashconfig=(-DDEFAULT_PATH_VALUE=\'\"/usr/local/sbin:/usr/local/bin:/usr/bin\"\'
+        -DSTANDARD_UTILS_PATH=\'\"/usr/bin\"\'
+        -DSYS_BASHRC=\'\"/etc/bash.bashrc\"\'
+        -DSYS_BASH_LOGOUT=\'\"/etc/bash.bash_logout\"\'
+        -DNON_INTERACTIVE_LOGIN_SHELLS)
+    export CFLAGS="${CFLAGS} ${_bashconfig[@]}"
+    export CFLAGS="${CFLAGS} -fprofile-generate -fprofile-update=atomic -fprofile-partial-training"
```
Comment (ptr1337) [inline bash-pgo/PKGBUILD:56]: Doesnt need a path defined for GCC PGO ?
like -fprofile-generate=$PWD/PGO

**Review by ptr1337 — COMMENTED** [review]
Please check the mentioned comments

File: bash-pgo/PKGBUILD [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+        --without-bash-malloc \
+        --with-installed-readline
+    make
+    make -C examples/loadables all others
+    make -k check
+
+    export CFLAGS="${CFLAGS//-fprofile-generate/}"
+    export CFLAGS="${CFLAGS//-fprofile-update=atomic/}"
+
+    export CFLAGS="${CFLAGS} -fprofile-use"
```
[the remark this answers is not in the document]
Comment (lseman) [reply bash-pgo/PKGBUILD:71]: No, for GCC the default profile folder is already the binary/library path, it generates the .cgda "next to" the binary.

File: bash-pgo/PKGBUILD (hunk 2 above)
Comment (lseman) [reply bash-pgo/PKGBUILD:56]: No, for GCC the default profile folder is already the binary/library path, it generates the .cgda "next to" the binary.

**Comment by lseman** [comment]
Ready to merge.

**Comment by lseman** [comment]
Please test in the cachybuilder.

**Review by ptr1337 — APPROVED** [review]
Seems good. tested locally

**Pull request merged**